spec.emu

```diff
@@ -1539,6 +1539,75 @@ return lastURL;</code></pre>
   </emu-clause>
 </emu-clause>
 
+<emu-clause id="sec-operations-on-source-map-records">
+  <h1>Operations on source map records</h1>
+
+  <p>After decoding a source map, source map consumers can use the resulting Decoded Source Map Records to look up position information for debugging or other use cases. This section describes the behaviour of typical operations that may be supported by source map consumers.</p>
+  <p>The GetOriginalPosition operation may be used to query the position in an original source that corresponds to a position in the generated code. For example, this can be used in a debugger to navigate from the generated code to the original source based on a user's mouse click.</p>
+  <p>The GetGeneratedPositionsForOriginalLine is used to query the positions in the generated code that map to a position in the original source. There may be multiple generated code positions that map to a given original position. The result list can be further filtered, for example, to match a particular column. This operation could be used, for example, as part of setting breakpoints at all generated lines that match a particular line in the original source that a user clicks in a debugger.</p>
+
+  <emu-clause id="sec-GetOriginalPosition" type="abstract operation">
+    <h1>
+      GetOriginalPosition (
+        _sourceMapRecord_: a Decoded Source Map Record,
+        _generatedPosition_: a Position Record,
+      ): a Decoded Mapping Record or *null*
+    </h1>
+    <dl class="header"></dl>
+    <emu-alg>
+      1. Let _mappings_ be _sourceMapRecord_.[[Mappings]].
+      1. Let _closest_ be *null*.
+      1. For each element _mapping_ of _mappings_, do
+        1. If the result of performing ComparePositions(_mapping_.[[GeneratedPosition]], _generatedPosition_) is ~greater~, then
+          1. Return _closest_.
+        1. Else,
+          1. Set _closest_ to _mapping_.
+      1. Return _closest_.
+    </emu-alg>
+  </emu-clause>
+
+  <emu-clause id="sec-GetGeneratedPositionsForOriginalLine" type="abstract operation">
+    <h1>
+      GetGeneratedPositionsForOriginalLine (
```

Comment:
`trace-mapping` [also has](https://github.com/jridgewell/sourcemaps/blob/c780a904904048808f6043969dc1b87fc5b361d1/packages/trace-mapping/src/trace-mapping.ts#L235-L253) a `generatedPositionFor`/`allGeneratedPositionFor` and match Chrome's behavior. I could add a column optional variation, but I'm curious what use case it solves.